### PR TITLE
GH-215 added toString() implementation

### LINKED ISSUE #215: Add toString() implementation to AppDeploymentRequest
Would be rather helpful while debugging

### PR DESCRIPTION
**Pull request by olegz**
- added toString() Implementation to AppDeploymentRequest
- polished/simplified some code

Resolves #215

### CODE REVIEW

File: spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/core/AppDeploymentRequest.java [hunk 1]
```diff
@@ -92,12 +94,8 @@ public AppDeploymentRequest(AppDefinition definition, Resource resource,
 		Assert.notNull(resource, "resource must not be null");
 		this.definition = definition;
 		this.resource = resource;
-		this.deploymentProperties = deploymentProperties == null
-				? Collections.<String, String>emptyMap()
-				: Collections.unmodifiableMap(deploymentProperties);
-		this.commandlineArguments = commandlineArguments == null
-				? Collections.<String>emptyList()
-				: Collections.unmodifiableList(commandlineArguments);
+		this.deploymentProperties = deploymentProperties;
```
Comment (ilayaperumalg) [inline spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/core/AppDeploymentRequest.java]: Isn't it still be possible to have `null` values for `deploymentProperties` and `commandlineArguments` when this constructor is invoked `directly` with `null` values?

Comment (olegz) [inline spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/core/AppDeploymentRequest.java]: Well if you look at the existing code the `null` was replaced with empty `List/Map`, so I've just eliminated the unnecessary IF

Comment (ilayaperumalg) [inline spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/core/AppDeploymentRequest.java]: But, what happens if this constructor `public AppDeploymentRequest(AppDefinition definition, Resource resource,Map<String, String> deploymentProperties, List<String> commandlineArguments)` is 
`directly` invoked with `null` deploymentProperties and commandlineArguments ?

Comment (olegz) [inline spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/core/AppDeploymentRequest.java]: @ilayaperumalg  I see your point, let me revise it a bit

**Comment by olegz** [comment]
@ilayaperumalg I've reverted constructor changes keeping `toString()` only. Let me know if you want me to squash it

**Comment by ilayaperumalg** [comment]
@olegz I can squash them during the merge.

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Squashed and merged as 47b2f6f

**Pull request closed without merging**